chebpy/core/decorators.py

```diff
@@ -87,9 +87,7 @@ def thewrapper(self, *args, **kwargs):
         xx = np.array([x]) if np.isscalar(x) else np.array(x)
         # discern between the array(0.1) and array([0.1]) cases
         if xx.size == 1:
-            try:
-                xx[0]
-            except:
+            if xx.ndim == 0:
                 xx = np.array([xx])
```

Comment:
Since the result of `xx[0]` is not stored, I assume the `try`-`catch` block is used to check if `xx` is scalar or array? In that case, the numpy built-in `ndim` _should_ do the trick.